chronosphere/resource_classic_dashboard.go

```diff
@@ -45,6 +47,99 @@ func resourceClassicDashboard() *schema.Resource {
 			StateContext: schema.ImportStatePassthroughContext,
 		},
 		CustomizeDiff: r.ValidateDryRun(&ClassicDashboardDryRunCount),
-		Schema:        tfschema.GrafanaDashboard,
+		Schema:        tfschema.ClassicDashboard,
 	}
 }
+
+// ClassicDashboardDryRunCount tracks how many times dry run is run during validation for testing.
+var ClassicDashboardDryRunCount atomic.Int64
+
+type classicDashboardConverter struct{}
+
+func (classicDashboardConverter) toModel(
+	d *intschema.ClassicDashboard,
+) (*models.Configv1GrafanaDashboard, error) {
+	dashMeta, dashboardJSON, err := ClassicDashboardExtract(d.DashboardJson)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the provided uid during create is empty, then the server
+	// will generate one (as uid mirrors slug). However, unlike
+	// slug, uid is embedded in dashboard_json and dashboard_json is
+	// not a computed field. To workaround this, the computed
+	// sub-fields of dashboard_json are manually stripped via a
+	// DiffSuppressFunc, which unfortunately means the server-side
+	// generated uid never makes it into the state file. Thus, if
+	// uid is missing, we must fallback to using the state file ID
+	// as a source-of-truth for slug.
+	if dashMeta.Slug == "" {
+		dashMeta.Slug = d.StateID
+	}
+
+	collSlug, collRef := collectionRefFromID(d.CollectionId.Slug())
+	return &models.Configv1GrafanaDashboard{
+		Name:           dashMeta.Name,
+		Slug:           dashMeta.Slug,
+		BucketSlug:     d.BucketId.Slug(),
+		CollectionSlug: collSlug,
+		Collection:     collRef,
+		DashboardJSON:  dashboardJSON,
+	}, nil
+}
+
+func (classicDashboardConverter) fromModel(
+	m *models.Configv1GrafanaDashboard,
+) (*intschema.ClassicDashboard, error) {
+	dashboardJSON, err := grafana.SanitizedDashboardJSON(string(m.DashboardJSON))
```

Comment:
nit: mind dropping `grafana` here too (maybe we can move it into this file if it's not used anywhere else)